couchbase/src/main/scala/akka/stream/alpakka/couchbase/scaladsl/DiscoverySupport.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.couchbase.scaladsl
+
+import java.util.concurrent.CompletionStage
+
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.discovery.Discovery
+import akka.stream.alpakka.couchbase.CouchbaseSessionSettings
+import akka.util.JavaDurationConverters._
+import com.typesafe.config.Config
+
+import scala.collection.immutable
+import scala.compat.java8.FunctionConverters._
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Utility to delegate Couchbase node address lookup to [[https://doc.akka.io/docs/akka/current/discovery/index.html Akka Discovery]].
+ */
+sealed class DiscoverySupport private {
+
+  /**
+   * Use Akka Discovery to read the addresses for `serviceName` within `lookupTimeout`.
+   */
+  private def readNodes(
+      serviceName: String,
+      lookupTimeout: FiniteDuration
+  )(implicit system: ActorSystem): Future[immutable.Seq[String]] = {
+    import system.dispatcher
+    val discovery = Discovery(system).discovery
+    discovery.lookup(serviceName, lookupTimeout).map { resolved =>
+      resolved.addresses.map(_.host)
```

Comment:
This is dropping the port and favoring host before `InetAddress`. I'm guessing the `port` could still be necessary?

Put another way, I think the typing of the return type could be stronger. OTOH, if there's a reason to drop all and just keep `host` then I think a comment with that reason would be great.

Reply:
I think the port isn't really used with Couchbase because it opens up a whole set of ports on each cluster node.